assets/js/modules/analytics-4/utils/withCustomDimensions.test.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * `withCustomDimensions` HOC tests.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+	provideUserAuthentication,
+	provideUserCapabilities,
+	render,
+} from '../../../../../tests/js/test-utils';
+import { ERROR_REASON_INSUFFICIENT_PERMISSIONS } from '../../../util/errors';
+import { MODULES_ANALYTICS_4 } from '../datastore/constants';
+import withCustomDimensions from './withCustomDimensions';
+
+describe( 'withCustomDimensions', () => {
+	const customDimension = 'test_custom_dimension';
+	const TestComponent = () => <div data-testid="component" />;
+	const WithCustomDimensionsComponent = withCustomDimensions( {
+		dimensions: [ customDimension ],
+	} )( TestComponent );
+
+	it( 'renders appropriate error if required custom dimensions are not available', () => {
+		const { container } = render( <WithCustomDimensionsComponent />, {
+			setupRegistry: ( registry ) => {
```

Comment:
I find it cleaner to initialize the test `registry` in a `beforeEach` and then pass it to in the render options rather than doing the other way around (we're too flexible! 😄)

This way we can't apply common setup (the way `beforeEach` is intended to be used).

In this case it might only be 
```js
provideUserAuthentication( registry );
provideUserCapabilities( registry );
```

but that does help keep each test a bit more concise.

Reply:
Updated, thank you!